docker/packager/centos/clickhouse.spec

```diff
@@ -0,0 +1,382 @@
+CLICKHOUSE_DATADIR=/var/lib/clickhouse
+CLICKHOUSE_LOGDIR=/var/log/clickhouse-server
+
+##
+##
+##
+function create_system_user()
+{
+    USER=$1
+    GROUP=$2
+    HOMEDIR=$3
+
+    echo "Create user ${USER}.${GROUP} with datadir ${HOMEDIR}"
+
+    # Make sure the administrative user exists
+    if ! getent passwd ${USER} > /dev/null; then
+        adduser \
+            --system \
+            --no-create-home \
+            --home ${HOMEDIR} \
+            --shell /sbin/nologin \
+            --comment "Clickhouse server" \
+            clickhouse > /dev/null
+    fi
+
+    # if the user was created manually, make sure the group is there as well
+    if ! getent group ${GROUP} > /dev/null; then
+        addgroup --system ${GROUP} > /dev/null
+    fi
+
+    # make sure user is in the correct group
+    if ! id -Gn ${USER} | grep -qw ${USER}; then
+        adduser ${USER} ${GROUP} > /dev/null
+    fi
+
+    # check validity of user and group
+    if [ "`id -u ${USER}`" -eq 0 ]; then
+        echo "The ${USER} system user must not have uid 0 (root). Please fix this and reinstall this package." >&2
+            exit 1
+    fi
+
+    if [ "`id -g ${GROUP}`" -eq 0 ]; then
+        echo "The ${USER} system user must not have root as primary group. Please fix this and reinstall this package." >&2
+            exit 1
+    fi
+}
+
+
+create_system_user $CLICKHOUSE_USER $CLICKHOUSE_GROUP $CLICKHOUSE_DATADIR
+
+# Ensure required folders are in place
+if [ ! -d ${CLICKHOUSE_DATADIR} ]; then
+    mkdir -p ${CLICKHOUSE_DATADIR}
+    chown ${CLICKHOUSE_USER}:${CLICKHOUSE_GROUP} ${CLICKHOUSE_DATADIR}
+    chmod 700 ${CLICKHOUSE_DATADIR}
+fi
+
+if [ ! -d ${CLICKHOUSE_LOGDIR} ]; then
+    mkdir -p ${CLICKHOUSE_LOGDIR}
+    chown root:${CLICKHOUSE_GROUP} ${CLICKHOUSE_LOGDIR}
+    # Allow everyone to read logs, root and clickhouse to read-write
+    chmod 775 ${CLICKHOUSE_LOGDIR}
+fi
+
+# Clean old dynamic compilation results
+if [ -d "${CLICKHOUSE_DATADIR}/build" ]; then
+    rm -f ${CLICKHOUSE_DATADIR}/build/*.cpp ${CLICKHOUSE_DATADIR}/build/*.so ||:
+fi
+
+%preun server
+if [ $1 = 0 ]; then
+    /sbin/service clickhouse-server stop > /dev/null 2>&1
+    /sbin/chkconfig --del clickhouse-server
+fi
+
+%postun server
+if [ $1 -ge 1 ]; then
+    /sbin/service clickhouse-server restart >/dev/null 2>&1
+fi
+
+
+##
+## test package
+##
+
+%files test
+/usr/bin/clickhouse-test
+/usr/bin/clickhouse-test-server
+/usr/bin/clickhouse-performance-test
+# folder
+/usr/share/clickhouse-test
+%config(noreplace) /etc/clickhouse-client/client-test.xml
+%config(noreplace) /etc/clickhouse-server/server-test.xml
+
+##
+##
+##
+
+
+## TODO:
+# proper versions by macroses from packager
+# service https://src.fedoraproject.org/rpms/postgresql/blob/master/f/postgresql.spec#_855
+# better user creation
+# check tests
+# docs
+# proper relative paths (https://docs.fedoraproject.org/en-US/packaging-guidelines/RPMMacros/)
+
+
+%changelog
+* Wed May 17 2017 Vadim Tkachenko
```

Comment:
My understanding is that this should be the changelog of Clickhouse, not of the spec file

Reply:
> The last section, %changelog is a list of date-stamped entries that correlate to a specific Version-Release of the package. This is not meant to be a log of what changed in the software from release to release, but specifically to packaging changes

Reply:
Indead, I guess I learn everyday :)